source/common/http/http3/conn_pool.cc

```diff
@@ -82,7 +85,12 @@ allocateConnPool(Event::Dispatcher& dispatcher, Random::RandomGenerator& random_
         data.connection_ =
             Quic::createQuicNetworkConnection(h3_pool->quicInfo(), pool->dispatcher(), host_address,
                                               source_address, quic_stat_names, scope);
-        return std::make_unique<ActiveClient>(*pool, data);
+        Network::Connection& connection = *data.connection_;
+        auto client = std::make_unique<ActiveClient>(*pool, data);
+        if (connection.state() == Network::Connection::State::Closed) {
```

Comment:
nit: Any reason not to just use data.connection_->state() here?

Reply:
It is moved in the construction.

Reply:
Oh, I see. It's too bad data isn't passed via std::move(data) to make this clear, but fair enough. Perhaps a comment that explains why we use a local variable?

Reply:
done